extensions/amp-wordpress-embed/1.0/component.js

```diff
@@ -0,0 +1,136 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '#preact';
+import {IframeEmbed} from '#preact/component/iframe';
+import {addParamToUrl} from '../../../src/url';
+import {dict} from '#core/types/object';
+import {forwardRef} from '#preact/compat';
+import {getData} from '../../../src/event-helper';
+
+const {useCallback, useEffect, useMemo, useRef, useState} = Preact;
+
+const NO_HEIGHT_STYLE = dict();
+
+/**
+ * @param {!WordPressEmbedDef.Props} props
+ * @param {{current: ?WordPressEmbedDef.Api}} ref
+ * @return {PreactDef.Renderable}
+ */
+function WordPressEmbedWithRef(
+  {children, requestResize, title = 'WordPressEmbed', url, ...rest},
+  ref
+) {
+  const [heightStyle, setHeightStyle] = useState(NO_HEIGHT_STYLE);
+  const [opacity, setOpacity] = useState(0);
+  const contentRef = useRef(null);
+  const [win, setWin] = useState(null);
+
+  const iframeURL = useMemo(() => {
+    return addParamToUrl(url, 'embed', 'true');
+  }, [url]);
+
+  const matchesMessagingOrigin = useCallback(
+    (testURL) => {
+      const embeddedUrl = new URL(url);
+      const checkedUrl = new URL(testURL);
+      return embeddedUrl.origin === checkedUrl.origin;
+    },
+    [url]
+  );
+
+  const messageHandler = useCallback(
+    (event) => {
+      const data = getData(event);
+
+      switch (data.message) {
+        case 'height':
+          if (typeof data.value === 'number') {
+            const height = data.value;
+            if (requestResize) {
+              requestResize(height);
+            }
+            setHeightStyle(dict({'height': height}));
+            setOpacity(1);
+          }
+          break;
+        case 'link':
+          // Only follow a link message for the currently-active iframe if the link is for the same origin.
+          // This replicates a constraint in WordPress's wp.receiveEmbedMessage() function.
+          if (matchesMessagingOrigin(data.value) && win) {
+            win.top.location.href = data.value;
+          }
+          break;
+      }
+    },
+    [requestResize, matchesMessagingOrigin, win]
+  );
+
+  useEffect(() => {
+    setWin(contentRef.current?.ownerDocument?.defaultView);
+  }, []);
+
+  // Checking for valid props
+  if (!checkProps(url)) {
+    return null;
+  }
+
+  return (
+    <>
+      <IframeEmbed
+        allowTransparency
+        iframeStyle={{opacity}}
+        matchesMessagingOrigin={matchesMessagingOrigin}
+        messageHandler={messageHandler}
+        ref={ref}
+        contentRef={contentRef}
+        src={iframeURL}
+        wrapperStyle={heightStyle}
+        title={title}
+        {...rest}
+      />
+      {children}
```

Comment:
What is an example of `children` that this component should expect? Do we expect publishers to provide them? I am wondering if this can be removed in both layers (here, and `props.children` entry from `BaseElement`). 

Note that AMP specific elements, such as placeholder, fallback, and overflow, are displayable regardless.

Reply:
@caroqliu the `children` is required for the `overflow` button and I've added examples in both [amp-wordpress-embed.md](https://github.com/ampproject/amphtml/pull/34948/files/6883738204da878d7418f7c4aa53e4674c7c2542#diff-253733eb9b30de47df4e2c93a894f252ae66adaefc9459ca54459ca862344099R42) and in [Basic.amp.js](https://github.com/ampproject/amphtml/pull/34948/files/6883738204da878d7418f7c4aa53e4674c7c2542#diff-0273d6327aaaf5f6e4f85d6418d76a988e45ef520e4f6fe6c90ca1441560c0e0R48) files.

Reply:
> Note that AMP specific elements, such as placeholder, fallback, and overflow, are displayable regardless.

I've tested in the storybook component and without the `children` the overflow button doesn't show up.

Reply:
Ah, this should be removable if you please merge with main to get #35003 in your branch.

Reply:
Thank you @caroqliu. I've removed the `children` from both the component and the `BaseElement`.